tests/mmm-mpl-core.spec.ts

```diff
@@ -152,73 +174,1251 @@ describe('mmm-mpl-core', () => {
+        .mplCoreFulfillSell({
+          maxPaymentAmount: new anchor.BN(
+            1.02 * LAMPORTS_PER_SOL + expectedTakerFees,
+          ),
+          buysideCreatorRoyaltyBp: 10000,
+          allowlistAux: '',
+          takerFeeBp: 100,
+          makerFeeBp: 0,
+          compressionProof: null,
+        })
+        .accountsStrict({
+          payer: buyer.publicKey,
+          owner: wallet.publicKey,
+          cosigner: cosigner.publicKey,
+          referral: poolData.referral.publicKey,
+          pool: poolData.poolKey,
+          asset: asset.publicKey,
+          collection: collection!.publicKey,
+          sellState,
+          buysideSolEscrowAccount,
+          systemProgram: SystemProgram.programId,
+          assetProgram: MPL_CORE_PROGRAM_ID,
+        })
+        .remainingAccounts([
+          {
+            pubkey: creator1.publicKey,
+            isSigner: false,
+            isWritable: true,
+          },
+          {
+            pubkey: creator2.publicKey,
+            isSigner: false,
+            isWritable: true,
+          },
+        ])
+        .preInstructions([
+          ComputeBudgetProgram.setComputeUnitLimit({
+            units: 1_000_000,
+          }),
+        ])
+        .transaction();
+
+      const blockhashData = await connection.getLatestBlockhash();
+      tx.feePayer = buyer.publicKey;
+      tx.recentBlockhash = blockhashData.blockhash;
+      tx.partialSign(cosigner, buyer);
+
+      await sendAndAssertTx(connection, tx, blockhashData, false);
+
+      const refreshedAssetAfterFulfill = await getTestMplCoreAsset(
+        asset.publicKey,
+      );
+
+      // Verify asset account.
+      assert.equal(
+        refreshedAssetAfterFulfill.owner.toString(),
+        buyer.publicKey.toString(),
+      );
+
+      // verify post balances
+      const [
+        buyerBalance,
+        sellerBalance,
+        creator1Balance,
+        creator2Balance,
+        buyerSolEscrowAccountBalance,
+      ] = await Promise.all([
+        connection.getBalance(buyer.publicKey),
+        connection.getBalance(wallet.publicKey),
+        connection.getBalance(creator1.publicKey),
+        connection.getBalance(creator2.publicKey),
+        connection.getBalance(buysideSolEscrowAccount),
+      ]);
+
+      const expectedTxFees = SIGNATURE_FEE_LAMPORTS * 2; // cosigner + payer
+      const sellStatePDARent = await getSellStatePDARent(connection);
+
+      assert.equal(
+        creator1Balance,
+        initCreator1Balance + 1 * LAMPORTS_PER_SOL * 0.02 * 0.3, // 30% of the royalty
+      );
+      assert.equal(
+        creator2Balance,
+        initCreator2Balance + 1 * LAMPORTS_PER_SOL * 0.02 * 0.7, // 70% of the royalty
+      );
+
+      assert.equal(
+        buyerBalance,
+        initBuyerBalance -
+          1.02 * LAMPORTS_PER_SOL -
+          expectedTxFees -
+          expectedTakerFees,
+      );
+      assert.equal(sellerBalance, initSellerBalance + sellStatePDARent);
```

Comment:
can we have a test where reinvest is false?

Reply:
yeah we already have one: https://github.com/magicoss/mmm/blob/37d0f8378b1a0f51e9a99ea4424120e638fe8286/tests/mmm-mpl-core.spec.ts#L1114